Bstein dev performance tweaks

bstein-clever opened the pull request:
**Overview:**
One of our workers was timing out after 12 hours - trying various improvements to speed things up.
This set of tweaks helps by:
- Using faster json encoding/decoding package
- Only requesting configured fields from Mongo.
- Using a faster compression scheme for the gzip files on s3.

The combination of these changes saves about 1/3 of the processing time, at least when run locally.

**Testing:**
Go profiling and running the tasks locally.

**Roll Out:**
Just deploy it.

bstein-clever commented:
It came recommended to me by @natebrennand, and mentioned on some blog posts, though I'm not sure offhand which ones.

natebrennand commented on main.go:
```diff
@@ -320,7 +328,7 @@ func main() {
 			reader, writer := io.Pipe()
 			go func(index int) {
-				zippedOutput := gzip.NewWriter(writer) // sorcery
+				zippedOutput, _ := gzip.NewWriterLevel(writer, gzip.BestSpeed) // sorcery
```
let's check this error

(or at least leave a comment that we don't need to check it because we're using their constant: https://golang.org/src/compress/gzip/gzip.go?s=1836:1896#L50)

I'd rather that we check it in case someone plays with the value at a later date

bstein-clever commented:
Just me failing on the dep add step, since it added it on the make install_deps. I added it to the toml.

I also added a commit for some mongo batch/prefetch stuff.
